corehq/apps/users/views/mobile/users.py

```diff
@@ -1126,6 +1127,60 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
+class DeleteCommCareUsers(BaseManageCommCareUserView):
+    urlname = 'delete_commcare_users'
+    page_title = ugettext_noop('Bulk Delete')
+    template_name = 'users/bulk_delete.html'
+
+    @property
+    def page_context(self):
+        context = self.main_context
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(),
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
+        try:
+            workbook = get_workbook(request.FILES.get('bulk_upload_file'))
+        except WorkbookJSONError as e:
+            messages.error(request, str(e))
+            return self.get(request, *args, **kwargs)
+
+        try:
+            sheet = workbook.get_worksheet()
+        except WorksheetNotFound:
+            messages.error(request, _("Workbook has no worksheets"))
+            return self.get(request, *args, **kwargs)
+
+        deleted_count = 0
+        usernames_with_forms = []
+        for row in sheet:
+            try:
+                username = row['username']
+            except KeyError:
+                messages.error(request, _("Please upload a file with a 'username' column."))
+                return self.get(request, *args, **kwargs)
+
+            user = CommCareUser.get_by_username(f"{username}@{request.domain}.commcarehq.org")
+            if user:
+                if FormES().domain(request.domain).user_id(user.get_id).count():
```

Comment:
could do a single request to check if users have forms by doing a terms aggregation on `username` or `user_id`:

```python
user_ids_with_forms = FormES().domain(request.domain).user_id(user_ids).terms_aggregation('user_id', 'user_id').run().aggregations.user_id..keys
```

Reply:
It doesn't seem to work the same, tried with a local user who has form submissions but the aggregate query doesn't return anything:
```
In [34]: FormES().domain(domain).user_id(user_id).count()
Out[34]: 4

In [35]: FormES().domain(domain).user_id([user_id]).terms_aggregation('user_id', 'user_id').run().aggregations.user_id.keys
Out[35]: []
```

Reply:
I think it needs to use [`form.meta.userID`](https://github.com/dimagi/commcare-hq/blob/3cc508d2668776578a17465fb7e8f5bf60226b87/corehq/apps/es/forms.py#L73-L79) as the user ID path.
```python
FormES().domain(domain).user_id([user_id]).terms_aggregation('form.meta.userID', 'user_id').run().aggregations.user_id.keys
```

Reply:
That fixes it, thank you!

Reply:
77f2d7d7be5

Reply:
77f2d7d7be5